File: Algorithm/Selectionsort.java
import java.io.IOException;

public class SelectionSort {
    public static void main(String[] args) throws IOException {
        // write your code here
        int[] arr = {3,5,2,6,4,7,9,1,8};

        for(int num: arr){
            System.out.printf("%d, ", num);
        }
        System.out.println();

        for(int i=0; i<arr.length; i++){
            int indexMin = i;
            for(int j=i+1; j<arr.length; j++){
                if(arr[j] < arr[indexMin]){
                    indexMin = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[indexMin];
            arr[indexMin] = temp;
        }

        for(int num: arr){
            System.out.printf("%d, ", num);
        }
    }
}

File: Algorithm/Udemy_Javascript/sort/selectionSort.js
// 배열에서 최소값의 인덱스를 찾아서 매 단계의 첫번째 요소와 스왑하는 알고리즘
function selectionSort(arr) {
  for (let i = 0; i < arr.length; i++) {
    let noSwaps = true;
    let lowest = i;
    for (let j = i + 1; j < arr.length; j++) {
      if (arr[lowest] > arr[j]) {
        lowest = j;
      }
    }
    if (lowest !== i) {
      [arr[i], arr[lowest]] = [arr[lowest], arr[i]];
      noSwaps = false;
    }
    if (noSwaps) break;
  }

  return arr;
}

console.log(selectionSort([27, 11, 87, 21, 5, 9, 16]));
